SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TriangleSetGeometryAlgorithms.inl

```diff
@@ -1623,7 +1633,313 @@ bool TriangleSetGeometryAlgorithms< DataTypes >::computeSegmentTriangleIntersect
     return is_validated;
 }
 
+// Computes the intersection of the segment from point a to point b and the triangle indexed by t
+template<class DataTypes>
+bool TriangleSetGeometryAlgorithms< DataTypes >::computeIntersectionsLineTriangle(bool is_entered,
+    const sofa::defaulttype::Vec<3, double>& a,
+    const sofa::defaulttype::Vec<3, double>& b,
+    const TriangleID ind_t,
+    sofa::helper::vector<TriangleID>& indices,
+    sofa::helper::vector<double>& vecBaryCoef,
+    sofa::helper::vector<double>& vecCoordKmin) const
+{
+    double baryCoef;
+    double coord_kmin;
+    // HYP : point a is in triangle indexed by t
+    // is_entered == true => indices.size() == 2
+    TriangleID ind_first = 0;
+    TriangleID ind_second = 0;
+
+    if (indices.size() > 1)
+    {
+        ind_first = indices[0];
+        ind_second = indices[1];
+    }
+
+    indices.clear();
+    vecBaryCoef.clear();
+
+    bool is_validated = false;
+    bool is_intersected = false;
+
+    const Triangle& t = this->m_topology->getTriangle(ind_t);
+    const typename DataTypes::VecCoord& vect_c = (this->object->read(core::ConstVecCoordId::position())->getValue());
+
+    bool is_full_01 = (is_entered && ((t[0] == ind_first && t[1] == ind_second) || (t[1] == ind_first && t[0] == ind_second)));
+    bool is_full_12 = (is_entered && ((t[1] == ind_first && t[2] == ind_second) || (t[2] == ind_first && t[1] == ind_second)));
+    bool is_full_20 = (is_entered && ((t[2] == ind_first && t[0] == ind_second) || (t[0] == ind_first && t[2] == ind_second)));
+
+    const typename DataTypes::Coord& c0 = vect_c[t[0]];
+    const typename DataTypes::Coord& c1 = vect_c[t[1]];
+    const typename DataTypes::Coord& c2 = vect_c[t[2]];
+
+    sofa::defaulttype::Vec<3, Real> p0;
+    p0[0] = (Real)(c0[0]);
+    p0[1] = (Real)(c0[1]);
+    p0[2] = (Real)(c0[2]);
+    sofa::defaulttype::Vec<3, Real> p1;
+    p1[0] = (Real)(c1[0]);
+    p1[1] = (Real)(c1[1]);
+    p1[2] = (Real)(c1[2]);
+    sofa::defaulttype::Vec<3, Real> p2;
+    p2[0] = (Real)(c2[0]);
+    p2[1] = (Real)(c2[1]);
+    p2[2] = (Real)(c2[2]);
+
+    sofa::defaulttype::Vec<3, Real> pa;
+    pa[0] = (Real)(a[0]);
+    pa[1] = (Real)(a[1]);
+    pa[2] = (Real)(a[2]);
+    sofa::defaulttype::Vec<3, Real> pb;
+    pb[0] = (Real)(b[0]);
+    pb[1] = (Real)(b[1]);
+    pb[2] = (Real)(b[2]);
+
+    sofa::defaulttype::Vec<3, Real> v_normal = (p2 - p0).cross(p1 - p0);
+    Real norm_v_normal = v_normal.norm(); // WARN : square root COST
+
+    if (norm_v_normal != 0.0)
```

Comment:
seems inconsistent that you dont use comparison with epsilon like in computeEdgeSegmentIntersection()